src/libraries/System.Net.Sockets/src/System/Net/Sockets/SocketAsyncEngine.Unix.cs

```diff
@@ -326,6 +287,59 @@ private void FreeNativeResources()
             }
         }
 
+        // The JIT is allowed to arbitrarily extend the lifetime of locals, which may retain SocketAsyncContext references,
+        // indirectly preventing Socket instances to be finalized, despite being no longer referenced by user code.
+        // To avoid this, the event handling logic is delegated to a non-inlined processing method.
+        // See discussion: https://github.com/dotnet/runtime/issues/37064
+        // SocketEventHandler holds an on-stack cache of SocketAsyncEngine members needed by the handler method.
+        private struct SocketEventHandler
```

Comment:
nit: it can be made `readonly`

```suggestion
        private readonly struct SocketEventHandler
```